---
title: "Distances, Trees & Types: Methods for Analyzing High-Dimensional Microbiome Data"
subtitle: "<hr>"
author: "Brendan J. Kelly, MD, MS"
institute: "Infectious Diseases, Epidemiology, & Microbiology" 
date: "17 June 2024" #"`r format(Sys.Date(), '%d %B %Y')`"
output:
  xaringan::moon_reader:
    css: xaringan-themer-inferno-grids-fade.css
    lib_dir: libs
    chakra: libs/remark-latest.min.js
    seal: false
    nature:
      ratio: 16:9
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      slideNumberFormat: "%current%"

---
exclude: true

```{r setup, include=FALSE, echo=FALSE, message=FALSE, error=FALSE, include=FALSE, comment=FALSE}
options(htmltools.dir.version = FALSE)
xaringan::summon_remark()
library(tidyverse)
library(gt)
library(gtsummary)

set.seed(16)

```

```{r xaringan-tile-view, echo=FALSE, message=FALSE, error=FALSE, include=FALSE, comment=FALSE}
xaringanExtra::use_tile_view()
xaringanExtra::use_webcam()
xaringanExtra::use_tachyons()
xaringanExtra::use_animate_css()
xaringanExtra::use_share_again()
#xaringanExtra::use_scribble(pen_color = "#56106E")
```

---
class: top left hide-count
background-image: url(img/adrian-lange-Wk902ZLaA7M-unsplash_crop.jpg)
background-size: cover

.move-top5[
.title-text[<span style='font-size:0.9em'>Distances, Trees & Types</span>]  
.title-subtext[<span style='font-size:0.95em'>Methods for Analyzing High-Dimensional Microbiome Data</span>]
]  
  
.callout-url-rightcorner[
.title-nametext[
Brendan J. Kelly, MD, MS  
Infectious Diseases | Epidemiology | Microbiology  
University of Pennsylvania  
17 June 2024  <!-- `r format(Sys.Date(), '%d %B %Y')` -->   
]
]

.footnote-left[<span style='color:#ffffff'>photo: Adrian Lange @unsplash</span>]  



---

## Disclosures  

.pad-left[  

- No conflicts of interest.  

- Research supported by:

    - NIAID K23 AI121485  
    - CDC BAA 200-2016-91964  
    - CDC BAA 200-2016-91937  
    - CDC BAA 200-2018-02919  
    - CDC BAA 200-2021-10986   
    - CDC Prevention Epicenters U54CK000610  
    - NIAID U19AI174998  

]

---

## Outline: The Microbiome for Epidemiologists   
  
.pull-left[  
  
- The problem: **too much** data

    - the "p > n problem"  

- Reducing dimensions:  

    - richness, evenness, and diversity  

    - ecological distances (UniFrac)  

    - PCA & PCoA  

    - PERMANOVA (adonis)  

]

.pull-right[

- Cluster analysis:  

    - supervised vs unsupervised learning  

    - Dirichlet multinomial mixture modeling

```{r, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}

knitr::include_graphics(path = "img/pptx_img/image25.png")

```


]  




---
class: top left hide-count
background-image: url(img/diana-polekhina-z84ULL2EoKs-unsplash.jpg)
background-size: cover

## <span style='color: #093b85;'>Too Much Data</span>


.footnote-left[<span style='color: #093b85'>photo: Diana Polekhina @unsplash</span>]  


---

## Generating Microbiome Data  

.pad-left[

- High-density (next-generation, high-throughput) sequencing:  

    - “tag” gene with conserved and variable regions (16S, 18S, ITS)  

    - "shotgun" metagenomics (pool of randomly amplified nucleic acids)  

- Sequence binning and assignment:  

     - operational taxonomic units (OTUs) based on 97% sequence similarity → taxonomic assignment of OTUs  

     - assemble contiguous metagenomic sequences → taxonomy

     - unassembled/un-binned reads → taxonomic assignment (e.g., ASVs or SGBs) 

]


.footnote-left[Callahan BJ _bioArxiv_ 2017]


---

## An Example OTU Table  

```{r, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}

set.seed(16)

# read counts
sample(x = seq(0,1000,1), size = 1e4, replace = TRUE, prob = c(0.8, rep(0.2 / 100, 1000))) |> 
   matrix(data = _, nrow = 1000, ncol = 10) |> 
   as.data.frame(row.names = paste0("OTU_",seq(1000))) |> 
   as_tibble(rownames = "OTU ID") |>
   rename_all(.funs = ~ gsub("V","Specimen_", .x)) |> 
   gt::gt_preview() |> 
   gt::as_raw_html()


```



---

## An Example OTU Table: Samples & Variables?    

```{r, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}

set.seed(16)

# read counts
sample(x = seq(0,1000,1), size = 1e4, replace = TRUE, prob = c(0.8, rep(0.2 / 100, 1000))) |> 
   matrix(data = _, nrow = 1000, ncol = 10) |> 
   as.data.frame(row.names = paste0("OTU_",seq(1000))) |> 
   as_tibble(rownames = "OTU ID") |>
   rename_all(.funs = ~ gsub("V","Specimen_", .x)) |> 
   gt::gt_preview() |> 
   gt::as_raw_html()


```

.pad-left[

- which are the samples and which are the variables?  

- also note: lots of zeros  

]


---

## An Example OTU Table: Samples vs Variables    

```{r, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}

set.seed(16)

# read counts
sample(x = seq(0,1000,1), size = 1e4, replace = TRUE, prob = c(0.8, rep(0.2 / 100, 1000))) |> 
   matrix(data = _, nrow = 1000, ncol = 10) |> 
   t() |> 
   as.data.frame(row.names = paste0("Specimen_",seq(10))) |> 
   as_tibble(rownames = "Specimen ID") |>
   rename_all(.funs = ~ gsub("V","OTU_", .x)) |> 
   gt::gt_preview() |> 
   gt::as_raw_html()


```

.pad-left[

- OTUs are really *columns* (i.e., variables) & specimens *rows* (i.e., observations)  

- 10 rows & 1000 columns → p >> n → overfitting risk   

]


---
class: top left hide-count
background-image: url(img/michael-48yI_ZyzuLo-unsplash.jpg)
background-size: cover

## <span style='color: black;'>Reducing Dimensions</span>


.footnote-right[<span style='color: white'>photo: Michael @unsplash</span>]  


---

## High-Dimensional Microbiome Data  

.pad-left[

- Descriptive:  

    - heatmaps  

    - stacked barplots  

- Test a priori hypotheses regarding specific OTUs/taxa.  

- Reduce dimensions:  

    - single summary statistic (alpha diversity)  

    - pairwise distances (beta diversity) with PCoA or PERMANOVA  

    - community types (mixture modeling)  

]

---
class: top left
background-image: url(img/pptx_img/image4.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image3.png)
background-size: contain


---

## Descriptive: Heatmaps & Barplots  

.pad-left[

- Visualization of OTU table:  

    - typically present counts as a proportion of sample total  

    - choice of sample order can highlight group differences  

- Limitations:  

    - cannot depict full list of OTUs  

    - space dictates taxonomic level presented  

]

---

## Single-Taxon Hypotheses

.pad-left[

- You suspect Bacteroides has a relationship with outcome of interest...  

    - Bacteroides (genus)?  

    - Bacteroidaceae (family)?  

    - Bacteroidales (order)?  

    - Bacteroidetes (class)?

    - functional group (e.g., butyrate production)?  

- Hypotheses focusing on specific taxa often fail to account for **possibility of selection bias from culture**.  


]

---

## Single-Taxon Hypotheses

```{r, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}

set.seed(16)

read_csv("tabs/demo_taxonomic_assignment_for_CCEB_ID_epi_lecture_20240616.csv") |> 
   select(-assignment) |> 
   gt::gt() |> 
   gt::as_raw_html()


```


---

## Dimension Reduction: Alpha Diversity  

.pad-left[

- Summarize each sample’s community in a single measure:  

    - richness: number of community members  

    - evenness: the distribution of member counts  

- Many alpha diversity metrics (weight richness/evenness):  

    - species number, Chao1 (singletons & doubletons)  

    - Shannon diversity: $H' = - \sum_{i} p_{i} \cdot \log(p_{i})$

     (note: may measure similarity or dissimilarity)  


]

---
class: top left
background-image: url(img/pptx_img/image2.png)
background-size: contain


---

## Dimension Reduction: Beta Diversity  

.pad-left[

- Summarize each sample’s relationship to other samples:  

    - pairwise distances  

    - OTU table → square matrix   

- Many beta diversity metrics:  

    - just counts versus counts + phylogeny  

    - weighted versus unweighted  
    
    - (Euclidean versus non-Euclidean)


]

---
class: top left
background-image: url(img/pptx_img/image19.png)
background-size: contain


.footnote-left[Anderson MJ et al. _Austral Ecol_ 2001;26(1):32-46.]

---

## Dimension Reduction: Beta Diversity  

.pad-left[

- Just counts versus counts + phylogeny:  

    - Jaccard: $J(A,B) = \frac{|A \cap B|}{|A \cup B|} = \frac{|A \cap B|}{|A| + |B| - |A \cap B|}$   

    - Jaccard distance: $d_{J}(A,B) = 1 - \frac{|A \cap B|}{|A \cup B|} = \frac{|A \cup B| - |A \cap B|}{|A \cup B|}$

    - UniFrac: fraction of unique branch length in tree  

- Weighted versus unweighted:  

    - weighted: counts matter  

    - unweighted: binary (presence-absence)  


]

---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='50%'}
knitr::include_graphics(path = "img/pptx_img/image5.png")

```

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='70%'}
knitr::include_graphics(path = "img/pptx_img/image7.png")

```


---

## Beta Diversity: How to Choose?  

.pad-left[

- Why use Jaccard? UniFrac?  

- Why use weighted distances? Unweighted distances?  


]


---

## Thinking Like a Tree  

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='48%'}
knitr::include_graphics(path = "img/pptx_img/image31.png")

```

.footnote-left[Johnson AJ et al. _Cell Host Microbe_ 2019]

---

## Dimension Reduction: PCA  

.pad-left[

- PCA: principal component analysis  

    - uses original descriptors (e.g., OTU abundance)  

    - rigid rotation for successive directions of maximum variance  

    - lots of restrictions (Euclidean)  

    - but allows projection of original descriptors in PCA space

]

---
class: top left
background-image: url(img/ijerph-19-10107-g003.jpg)
background-size: contain

.footnote-left[Perrone MR et al _Int J Environ Res Public Health_ 2022]

---

## Dimension Reduction: PCoA  

.pad-left[

- PCoA: principal coordinate analysis  

    - uses pairwise distances  

    - any metric distance, even if non-Euclidean  

    - like PCA, eigenvalue decomposition (maximum variance) but mediated by distance function (no original descriptors)  

    - unlike PCA, does not allow projection of original descriptors in reduced-dimension space  

]

---
class: top left
background-image: url(img/pptx_img/image9.png)
background-size: contain

.footnote-left[Legendre & Legendre _Numerical Ecology_ 3rd edition]


---
class: top left
background-image: url(img/pptx_img/image18.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image26.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image8.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image35.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image10.png)
background-size: contain


---

## Dimension Reduction: PERMANOVA (adonis)  

.pad-left[

- Pairwise distance matrix can be partitioned by group assignment and ANOVA-like analysis can be applied to detect difference between groups.  

- PERMANOVA: permutational ANOVA (aka, adonis)  

    - pseudo F-ratio: conceptually similar but not F-distributed  

    - testing by label permutation  

    - quantification of effect size by R-squared or omega-squared (the latter a less biased estimator of true effect)  


]

---
class: top left
background-image: url(img/pptx_img/image19.png)
background-size: contain


---

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='90%'}
knitr::include_graphics(path = "img/pptx_img/image16.png")

```

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='40%'}
knitr::include_graphics(path = "img/pptx_img/image17.png")

```


---
class: top left
background-image: url(img/pptx_img/image13.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image12.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image14.png)
background-size: contain


---
class: top left
background-image: url(img/pptx_img/image15.png)
background-size: contain


---

## PERMANOVA: Effect Sizes  

.pull-left[

$$R^{2} = 1 - \frac{SS_{W}}{SS_{W} + SS_{A}} = \frac{SS_{A}}{SS_{T}}$$

]

.pull-right[

$$\omega^{2} = \frac{SS_{A} - (a - 1) \cdot \frac{SS_{W}}{N - a}}{SS_{T} + \frac{SS_{W}}{N - a}}$$

]

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='80%'}
knitr::include_graphics(path = "img/pptx_img/image21.png")

```


.footnote-left[Kelly BJ et al. Bioinformatics 2015;31(15):2461-8]

---
class: top left
background-image: url(img/pptx_img/image32.png)
background-size: contain


.footnote-left[Kelly BJ et al. Bioinformatics 2015;31(15):2461-8]


---
class: top left hide-count
background-image: url(img/christina-victoria-craft-WHSnkIwWpec-unsplash.jpeg)
background-size: cover

## <span style='color: #226252'>The LBP Era</span>  


.footnote-right[<span style='color: white'>photo: Christina Victoria Craft @unsplash</span>]  



---
class: middle right hide-count
background-image: url(img/Gemini_Generated_Image_i7lg9vi7lg9vi7lg_crop.jpg)
background-size: cover

## <span style='color: white;'>Future Considerations</span>


.footnote-left[<span style='color: white'>photo: Google Gemini</span>]  


---

## MBT Efficacy Beyond R-CDI & Beyond the Gut  

.pull-left[

```{r, echo=FALSE, echo=FALSE, message=FALSE, warning=FALSE, error=FALSE, fig.align='center', out.width='100%'}
knitr::include_graphics(path = "img/otto_bacillus_effect.png")

```

]

.pull-right[

- MDRO pathogen burden reduction:   

    - FMT reduces gut reservoir of MDROs by strain replacement  

- MRSA decolonization:

   - oral probiotic _B. subtilis_ MB40 reduces not just enteric but also nasal reservoirs of MRSA  

]


.footnote-left[Gopalsamy et al _ICHE_ 2018; Saha et al _CMI_ 2019; Woodworth et al _STM_ 2023; Piewngam et al _Lancet Microbe_ 2023]  


---

## Acknowledgements    

.pull-left[
- __MOTION Lab & C-STAR @ Penn__
Laura Cowden, Laurel Glaser, Kyle Rodino, Sean Loughrey, 
Magda Wernovsky, Erik Clarke, Michael David, Matt Ziegler, 
Lauren Dutcher, Ebbing Lautenbach, Jim Harrigan, Alexa Patel, Dylan Tapper, Anne Jaskowiak, Pam Tolomeo   

- __Bushman Laboratory @ Penn__
Arwa Abbas, Aoife Roche, Andrew Marques,
Aubrey Bailey, John Everett,
Rick Bushman

- __PennCHOP Microbiome Program__
Lisa Mattei, Huanjia Zhang, Kyle Bittinger

]


.pull-right[
- __Collman Laboratory @ Penn__
Ize Imai, Aurea Simon Soro, John McGinniss, Ron Collman

- __Division of ID @ Penn__
Ian Frank, Pablo Tebas, Robert Gross, Emily Blumberg

- __Rutgers University & Penn DBEI__
Jason Roy, Arman Oganisian
  
- __AGA FMT National Registry__
Gary Wu, Colleen Kelly, Jim Lewis
  
- __Funding__
<u>CDC</u>: BAAs 200-2016-91964, 200-2018-02919, 200-2021-10986 & <u>NIAID</u>: K23 AI121485, U19AI174998

]

.center[
## brendank@upenn.edu
]



---
class: bottom center hide-count
background-image: url(img/adrian-lange-Wk902ZLaA7M-unsplash_crop.jpg)
background-size: cover

.title-text[<span style='font-size:0.5em; color:#ffffff'>Questions?</span>]  

.footnote-right[<span style='color:#ffffff'>photo: Adrian Lange @unsplash</span>]  


